bundle/jsky.app.ot/src/main/scala/jsky/app/ot/tpe/feat/TpeEphemerisFeature.scala

```diff
@@ -85,17 +95,19 @@ class TpeEphemerisFeature extends TpeImageFeature("Ephemeris", "Show interpolate
 
     // The main ephemeris track
     val se = toScreenEphemeris(e)
-    drawTrack (g, se)
-    drawLabels(g, se)
+    se.foreach(drawTrack(g, _))
+    se.foreach(drawLabels(g, _))
 
     // Highlighted track for remainder of observation, and label at the last point
     // on the highlighted track.
     obsScreenEphemeris(e).foreach { se =>
       g.setColor(Color.GREEN)
-      drawTrack(g, se)
-      se.takeRight(2) match {
-        case List((_, p1), (t, p2)) => drawLabel(g, t, p2, ang(p1, p2))
-        case _ => // degenerate ephemeris; no track was drawn, ignore
+      se.foreach(drawTrack(g, _))
+      se.foreach { lst =>
+        lst.takeRight(2) match {
+          case List((_, p1), (t, p2)) => drawLabel(g, t, p2, ang(p1, p2))
```

Comment:
````scala
type ScreenEphemeris = List[(Long, Point)]
````

So it is a `List` of times and the corresponding location on the screen.  `onScreenEphemeris` was modified to return a `List[ScreenEphemeris]` which is then a `List[List[(Long, Point)]]`. Each `ScreenEphemeris` entry in the list represents a continuous section of ephemeris elements for which we have a valid screen location.